sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -229,7 +229,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    */
   def load(path: String): DataFrame = {
     // force invocation of `load(...varargs...)`
-    option("path", path).load(Seq.empty: _*)
```

Comment:
To keep the behavior, I think we should still generate the `path` option when only one path is given in the parameters. We need to update L257.

Reply:
If we update L257, it will only handle the case when `DataSource.lookupDataSourceV2` returns `Some`?

Reply:
I pushed a new commit that doesn't change existing code except for the check. Please let me know what you think. Thanks!

Reply:
This works, another probably cleaner way is
```
def load(paths: String*) = {
  val pathOption = if (paths.length == 1) {
    "path" -> ...
  } else if (paths.length > 1) {
    "paths" ->
  } else None
}
```

Reply:
Sorry, I didn't get the suggestion. Are you suggesting not to check if the option already exists?  Then we still have to go [this](https://github.com/apache/spark/pull/29328/commits/63f3d60b0b00990cd5a2784b1be8961c058dc2c6#diff-f70bda59304588cc3abfa3a9840653f4R248) route where `paths` parameter is also updated, right? (The root cause of the inconsistency is that if you call `parquet(path)`, it would go to `load(paths: String*)` instead of `load(path: String)`.

Reply:
I mean, we can remove `def load(path: String): DataFrame` and only keep `def load(paths: String*): DataFrame`.

The check should be put in `def load(paths: String*)`, which should generate "path" option if `paths` only have one element.

Reply:
> The check should be put in `def load(paths: String*)`, which should generate "path" option if `paths` only have one element.

Generating a "path" option (assuming you meant `option("path", paths.head)`) is fine, but we need to set `paths` to `Seq.empty` since `loadV1Source` internally depends on the "path" option (e.g. `DataSource.checkAndGlobPathIfNecessary`). Then this becomes similar to [this](https://github.com/apache/spark/pull/29328/commits/63f3d60b0b00990cd5a2784b1be8961c058dc2c6#diff-f70bda59304588cc3abfa3a9840653f4R248)?

Reply:
> I mean, we can remove `def load(path: String): DataFrame` and only keep `def load(paths: String*): DataFrame`. (if it doesn't break binary compatibility)

Removing `def load(path: String)` breaks the binary compatibility:
Exception in thread "main" java.lang.NoSuchMethodError: org.apache.spark.sql.DataFrameReader.load(Ljava/lang/String;)Lorg/apache/spark/sql/Dataset;
So, I will not remove it.

Reply:
Instead of removing, I changed it to be just a passthrough so that we have a single place to check.